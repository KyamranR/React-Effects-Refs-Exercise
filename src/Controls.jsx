function Controls({ onDraw, onShuffle, isShuffling, autoDraw }) {
  return (
    <div>
      <button onClick={onDraw} disabled={isShuffling}>
        {autoDraw ? "Stop Drawing" : "Start Drawing"}
      </button>
      <button onClick={onShuffle} disabled={isShuffling || autoDraw}>
        {isShuffling ? "Shuffling..." : "Shuffle Deck"}
      </button>
    </div>
  );
}

export default Controls;
