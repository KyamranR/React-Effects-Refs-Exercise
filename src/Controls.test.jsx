import { render } from "@testing-library/react";
import Controls from "./Controls";

test("renders without crashing", () => {
  render(<Controls />);
});

test("matches snapshot", () => {
  const { asFragment } = render(<Controls />);
  expect(asFragment()).toMatchSnapshot();
});
