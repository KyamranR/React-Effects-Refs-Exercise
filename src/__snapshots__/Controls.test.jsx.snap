// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <button>
      Start Drawing
    </button>
    <button>
      Shuffle Deck
    </button>
  </div>
</DocumentFragment>
`;
